packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -0,0 +1,259 @@
+  children?: string;
+  isDisabled: boolean;
+  theme: Theme;
+  size: BaseButtonCommonProps['size'];
+}): BaseButtonStyleProps => {
+  const props: BaseButtonStyleProps = {
+    iconSize: 'medium',
+    fontSize: buttonTypographyTokens.fonts.size.medium,
+    lineHeight: buttonTypographyTokens.lineHeights.medium,
+    minHeight: '40px',
+    iconSpacing: makeSpace(theme.spacing[2]),
+    iconColor: 'action.icon.primary.default',
+    textColor: 'action.text.primary.default',
+    spacing: `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[5])}`,
+    text: children?.trim(),
+    color: getIn(theme.colors, 'action.background.primary.default'),
+    borderColor: getIn(theme.colors, 'action.border.primary.default'),
+    hoverColor: getIn(theme.colors, 'action.background.primary.hover'),
+    hoverBorderColor: getIn(theme.colors, 'action.border.primary.hover'),
+    activeColor: getIn(theme.colors, 'action.background.primary.active'),
+    activeBorderColor: getIn(theme.colors, 'action.border.primary.active'),
+    focusColor: getIn(theme.colors, 'action.background.primary.focus'),
+    focusBorderColor: getIn(theme.colors, 'action.border.primary.focus'),
+    focusRingColor: getIn(theme.colors, 'brand.primary.400'),
+    borderWidth: makeBorderSize(theme.border.width.thin),
+    borderRadius: makeBorderSize(theme.border.radius.small),
+  };
+
+  switch (size) {
+    case 'large':
+      props.iconSize = 'medium';
+      props.fontSize = buttonTypographyTokens.fonts.size.large;
+      props.lineHeight = buttonTypographyTokens.lineHeights.large;
+      props.minHeight = '48px';
+      props.iconSpacing = makeSpace(theme.spacing[2]);
+      props.spacing = `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[5])}`;
+      break;
+    case 'medium':
+      props.iconSize = 'medium';
+      props.fontSize = buttonTypographyTokens.fonts.size.medium;
+      props.lineHeight = buttonTypographyTokens.lineHeights.medium;
+      props.minHeight = '40px';
+      props.iconSpacing = makeSpace(theme.spacing[2]);
+      props.spacing = `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[5])}`;
+      break;
+    case 'small':
+      props.iconSize = 'xsmall';
+      props.fontSize = buttonTypographyTokens.fonts.size.small;
+      props.lineHeight = buttonTypographyTokens.lineHeights.small;
+      props.minHeight = '32px';
+      props.iconSpacing = makeSpace(theme.spacing[1]);
+      props.spacing = `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[3])}`;
+      break;
+    case 'xsmall':
+      props.iconSize = 'xsmall';
+      props.fontSize = buttonTypographyTokens.fonts.size.xsmall;
+      props.lineHeight = buttonTypographyTokens.lineHeights.xsmall;
+      props.minHeight = '28px';
+      props.iconSpacing = makeSpace(theme.spacing[1]);
+      props.spacing = `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[2])}`;
+      props.text = children?.trim().toUpperCase();
+      break;
+    default:
+  }
+
+  if (isDisabled) {
```

Comment:
In here we are natively passing the html `disabled` prop to be true when isDisabled is true right? @chaitanyadeorukhkar?? In that case focus styles won't even apply because focus will be lost.

Reply:
Yup thats correct. Focus styles won't apply when disabled is true